service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/AdditionalConfiguration.scala

```diff
@@ -23,8 +26,11 @@ trait ProvidesAdditionalConfiguration {
    *
    * When overriding, the overridden file should be a def, so as to ensure multiple components can all override it.
    * Lagom will only invoke this method once from a lazy val, so it will effectively be calculated once.
+   *
+   * @param initialConfiguration A reference to the application configuration prior to override.
    */
-  def additionalConfiguration: AdditionalConfiguration = AdditionalConfiguration.empty
+  def additionalConfiguration(initialConfiguration: Configuration): AdditionalConfiguration =
+    AdditionalConfiguration.empty
```

Comment:
This is a nice improvement. I wonder if instead of passing `initialConfig` we should pass the whole `playContext` or at least the `environment`.

Reply:
one of the things i was trying to avoid was having to do `Configuration.load(environment) ++ context.playContext.initialConfiguration` multiple times, but we could certainly pass the `environment` and/or `playContext` in addition to the initial configuration.

Reply:
I have decided to pass the `environment` in there in addition to `initialConfiguration` -- I left out the entire `playContext` because I feel like it may widen the use-cases of this method too far (for example, having access to application lifecycle hooks).